x-pack/plugins/security_solution/common/detection_engine/diffable_rule_to_rule_response.ts

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type * as t from 'io-ts';
+import { parseDuration } from '@kbn/alerting-plugin/common/parse_duration';
+
+import type {
+  DiffableRule,
+  DiffableCustomQueryFields,
+  DiffableSavedQueryFields,
+  DiffableEqlFields,
+  DiffableThreatMatchFields,
+  DiffableThresholdFields,
+  DiffableMachineLearningFields,
+  DiffableNewTermsFields,
+} from '../api/detection_engine/prebuilt_rules/model/diff/diffable_rule/diffable_rule';
+import type {
+  RuleSchedule,
+  SavedKqlQuery,
+  RuleDataSource as DiffableRuleDataSource,
+  RuleKqlQuery as DiffableRuleKqlQuery,
+} from '../api/detection_engine/prebuilt_rules/model/diff/diffable_rule/diffable_field_types';
+import type {
+  RuleResponse,
+  querySchema,
+  savedQuerySchema,
+  eqlSchema,
+  threatMatchSchema,
+  thresholdSchema,
+  machineLearningSchema,
+  newTermsSchema,
+  SharedResponseProps,
+  KqlQueryLanguage,
+} from '../api/detection_engine/model/rule_schema/rule_schemas';
+import type { RuleFilterArray } from '../api/detection_engine/model/rule_schema/common_attributes';
+import { assertUnreachable } from '../utility_types';
+
+type RuleResponseCustomQueryFields = t.TypeOf<typeof querySchema.create>;
+type RuleResponseSavedQueryFields = t.TypeOf<typeof savedQuerySchema.create>;
+type RuleResponseEqlFields = t.TypeOf<typeof eqlSchema.create>;
+type RuleResponseThreatMatchFields = t.TypeOf<typeof threatMatchSchema.create>;
+type RuleResponseThresholdFields = t.TypeOf<typeof thresholdSchema.create>;
+type RuleResponseMachineLearningFields = t.TypeOf<typeof machineLearningSchema.create>;
+type RuleResponseNewTermsFields = t.TypeOf<typeof newTermsSchema.create>;
+
+interface RuleResponseScheduleFields {
+  from: string;
+  to: string;
+  interval: string;
+}
+
+const extractRuleSchedule = (ruleSchedule: RuleSchedule): RuleResponseScheduleFields => {
+  const { interval, lookback } = ruleSchedule;
+  const lookbackSeconds = Math.floor(parseDuration(lookback) / 1000);
+  const intervalSeconds = Math.floor(parseDuration(interval) / 1000);
+  const totalSeconds = lookbackSeconds + intervalSeconds;
+
+  let totalDuration: string;
+  if (totalSeconds % 3600 === 0) {
```

Comment:
For this calculation, I saw we have a similar function. Can we reuse them, or are they different?

Like `getHumanizedDuration`?

Reply:
Yeah, I also thought about it, but it turned out that `getHumanizedDuration` is a little different - it accepts a different format of durations.